app/database/operator/server_data_operator/handlers/posts_in_channel.ts

```diff
@@ -98,6 +98,13 @@ const PostsInChannelHandler = (superclass: any) => class extends superclass {
         }
 
         if (targetChunk) {
+            if (
+                targetChunk.earliest <= earliest &&
+                targetChunk.latest >= latest
+            ) {
+                return [];
+            }
+
```

Comment:
For my understanding - what does this take care of ?

Reply:
This function creates/updates the needed records to the database, and return the modified models.

This is making sure that it only modify the models if they have to be modified. If not, return the empty list (no models has been modified). The same for your next question.

We do this so we don't make unneeded changes to the database, so there is less range of error for "updating the same model twice" errors.